File: feb07/src/add_client.py
#!/usr/bin/env python3
#coding:utf-8

import rospy
from feb07.srv import *

def client(a,b):
    rospy.init_node("add_client")
    rospy.loginfo("a = %lf, b = %lf",a,b)
    rospy.wait_for_service("add_two_number_py")
    try:
        test_client = rospy.ServiceProxy("add_two_number_py",test)
        resp = test_client(a,b)
        rospy.loginfo("The response is %lf", resp.c)
    except rospy.ServiceException:
        rospy.logwarn("Service call failde")


if __name__=="__main__":
    a = 1
    b = 2
    client(a,b);

File: feb07/src/add_client.cpp
#include <ros/ros.h>
#include <feb07/test.h>

int main(int argc, char **argv)
{
    ros::init(argc,argv,"add_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<feb07::test>("add_two_number");
    feb07::test srv;
    while(ros::ok()){
        std::cout<<"a = ";
        std::cin>>srv.request.a;
        std::cout<<"b = ";
        std::cin>>srv.request.b;

        if(client.call(srv)){
            std::cout<<"The response is "<<srv.response.c<<"\n";
        }
        else{
            std::cout<<"Failed to call service\n";
        }
    }
    return 0;
}